lib/membrane_http_adaptive_stream/stream_sink_bin.ex

```diff
@@ -0,0 +1,146 @@
+defmodule Membrane.HTTPAdaptiveStream.StreamSinkBin do
+  @moduledoc """
+  This bin takes a parsed video stream and dumps an HLS playlist produced from it to storage.
+
+  The bin payloads and muxes the incoming stream to the CMAF format and dumps it to an HLS playlist.
+
+  # Input streams
+  ! can this bin handle multiple streams??
+  The bin expects parsed H264 or AAC video streams to be connected via `:input` pads.
+  The type of video stream has to be specified via the `:encoding` option.
+
+  # Output
+  Specify one of `Membrane.HTTPAdaptiveStream.Storages` as `:storage` to configure the sink.
+
+
+  #! generalnie to nie wiem jak pisać dokumentację
+  """
+  use Membrane.Bin
+
+  alias Membrane.{ParentSpec, Time, MP4}
+  alias Membrane.HTTPAdaptiveStream.{Sink, Storage}
+
+  def_options muxer_segment_duration: [
+                type: :time,
```

Comment:
Don't use `type:`, this option should rather be deprecated. Please use `spec: {here goes type spec}` instead in every place.